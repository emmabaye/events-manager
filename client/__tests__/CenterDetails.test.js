import ConnectedCenterDetails, { CenterDetails } from '../components/CenterDetails.jsx';
import { BrowserRouter } from 'react-router-dom';

describe('CenterDetails Component', () => {
  const initialState = {
    centerReducer: {
      status: "Error",
      message: "",
      center: {
        data: [{
          id: '1',
          name: 'City Hall',
          description: {
            substr: () => {}
          },
          location: {
            substr: () => {}
          },
          capacity: "500",
          price: "50000",
          image: "",
          available: 'false',
          Events: []
        }]
      }
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const props = {
    dispatch: () => {},
    centerId: "1",
    center: initialState.centerReducer.center,
    match: {
      params: {
        id: '1'
      }
    },
  };

  describe('CenterDetails form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<BrowserRouter><CenterDetails {...props} /></BrowserRouter>);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<BrowserRouter><Provider store={store}><ConnectedCenterDetails match={props.match}/></Provider></BrowserRouter>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
