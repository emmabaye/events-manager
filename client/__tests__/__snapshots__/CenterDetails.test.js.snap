// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterDetails Component CenterDetails form  should render self it should render for dumb component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <CenterDetails
    center={
      Object {
        "data": Array [
          Object {
            "Events": Array [],
            "available": "false",
            "capacity": "500",
            "description": Object {
              "substr": [Function],
            },
            "id": "1",
            "image": "",
            "location": Object {
              "substr": [Function],
            },
            "name": "City Hall",
            "price": "50000",
          },
        ],
      }
    }
    centerId="1"
    dispatch={[Function]}
    match={
      Object {
        "params": Object {
          "id": "1",
        },
      }
    }
  />
</Router>
`;
